org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/SonarLintUiPlugin.java

```diff
@@ -181,9 +189,22 @@ public IStatus runInUIThread(IProgressMonitor monitor) {
       // Handle future opened/closed windows
       SonarLintUiPlugin.getDefault().getWorkbench().addWindowListener(new WindowOpenCloseListener());
 
+      analyzeCurrentFile();
+
       return Status.OK_STATUS;
     }
 
+    private static void analyzeCurrentFile() {
+      IWorkbenchPart workbenchPart = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage().getActivePart();
+      IEditorPart editor = workbenchPart.getSite().getPage().getActiveEditor();
+      if (editor != null) {
+        // note: the cast is necessary for e43 and e44
+        IFile file = (IFile) editor.getEditorInput().getAdapter(IFile.class);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this unnecessary cast to "IFile". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1905)